File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/JobApplication.cs
using System;
using HRManagementSystem.Domain.Enums;

namespace HRManagementSystem.Domain.Entities;

public class JobApplication : BaseEntity
{
    public int CandidateId { get; set; }
    public int JobPostingId { get; set; }
    public DateTime ApplicationDate { get; set; }
    public ApplicationStatus Status { get; set; }
    public ApplicationSource Source { get; set; }
    public string? CoverLetter { get; set; }
    public string? Notes { get; set; }
    public int? ReviewedBy { get; set; }
    public DateTime? ReviewedDate { get; set; }
    public DateTime? InterviewDate { get; set; }
    public string? InterviewFeedback { get; set; }
    public int? InterviewRating { get; set; } // 1-5 or 1-10 scale
    public decimal? ExpectedSalary { get; set; }
    public decimal? OfferedSalary { get; set; }
    public string? RejectionReason { get; set; }
    public int? AssignedRecruiterId { get; set; }
    public RecruitmentStage CurrentStage { get; set; }

    // Navigation properties
    public Candidate? Candidate { get; set; }
    public JobPosting? JobPosting { get; set; }
    public Employee? Reviewer { get; set; }
    public Employee? AssignedRecruiter { get; set; }
}

File: frontend/src/app/models/recruitment/job-application.ts
import { Candidate, CreateCandidateDto } from './candidate';
import { JobPosting } from './job-posting';

export interface JobApplication {
  id: number;
  candidateId: number;
  jobPostingId: number;
  applicationDate: string;
  status: string;
  source: string;
  coverLetter?: string;
  notes?: string;
  reviewedBy?: number;
  reviewedDate?: string;
  interviewDate?: string;
  interviewFeedback?: string;
  interviewRating?: number;
  expectedSalary?: number;
  offeredSalary?: number;
  rejectionReason?: string;
  assignedRecruiterId?: number;
  currentStage: string;
  createdAt: string;
  updatedAt?: string;
}

export interface JobApplicationDetail extends JobApplication {
  candidate: Candidate;
  jobPosting: JobPosting;
}

export interface CreateJobApplicationDto {
  candidateId?: number;
  candidateInfo?: CreateCandidateDto;
  jobPostingId: number;
  source: string;
  coverLetter?: string;
  expectedSalary?: number;
  notes?: string;
  assignedRecruiterId?: number;
}

export interface UpdateJobApplicationStatusDto {
  jobApplicationId: number;
  status: string;
  currentStage?: string;
  notes?: string;
  reviewedBy?: number;
  reviewedDate?: string;
  interviewDate?: string;
  interviewFeedback?: string;
  interviewRating?: number;
  rejectionReason?: string;
  offeredSalary?: number;
}

export enum ApplicationStatus {
  Applied = 'Applied',
  UnderReview = 'UnderReview',
  Shortlisted = 'Shortlisted',
  Interviewed = 'Interviewed',
  Offered = 'Offered',
  Accepted = 'Accepted',
  Rejected = 'Rejected',
  Withdrawn = 'Withdrawn',
}

export const ApplicationStatusLabels: Record<string, string> = {
  [ApplicationStatus.Applied]: 'Applied',
  [ApplicationStatus.UnderReview]: 'Under Review',
  [ApplicationStatus.Shortlisted]: 'Shortlisted',
  [ApplicationStatus.Interviewed]: 'Interviewed',
  [ApplicationStatus.Offered]: 'Offered',
  [ApplicationStatus.Accepted]: 'Accepted',
  [ApplicationStatus.Rejected]: 'Rejected',
  [ApplicationStatus.Withdrawn]: 'Withdrawn',
};

export const ApplicationStatusColors: Record<string, string> = {
  [ApplicationStatus.Applied]: 'bg-secondary',
  [ApplicationStatus.UnderReview]: 'bg-info',
  [ApplicationStatus.Shortlisted]: 'bg-primary',
  [ApplicationStatus.Interviewed]: 'bg-warning text-dark',
  [ApplicationStatus.Offered]: 'bg-success',
  [ApplicationStatus.Accepted]: 'bg-success',
  [ApplicationStatus.Rejected]: 'bg-danger',
  [ApplicationStatus.Withdrawn]: 'bg-dark',
};

export const ApplicationSources = [
  'LinkedIn',
  'Indeed',
  'Company Website',
  'Referral',
  'Job Fair',
  'Recruitment Agency',
  'Other',
];
